design/design.md

```diff
@@ -0,0 +1,414 @@
+
+Cross-chain bridges are protocols allowing one blockchain to verify
+information on another blockchain.
+In essense this is the same problem that the Mithril solves with a potentially
+added constraint that on-chain verification must be computationally cheap.
+Whereas Mithril is specific to the Cardano blockchain, one can envision
+a similar protocol, powered by ALBA, for other cryptocurrencies including
+partner chains like Midnight.
+
+Applications of ALBA are not limited to Cardano and blockchains in general.
+For instance, the [paper](https://eprint.iacr.org/2023/1655) describes how
+this primitive can be used to optimize the prover running time of
+a Universally Composable SNARK.
+
+## Requirements
+
+We intend to make an implementation of ALBA that can be used in
+Mithril, Peras, Leios, cross-chain bridges and potentially other protocols
+inside and outside the Cardano universe.
+Our implementation will mostly be guided by our main "clients":
+Mirthril, Peras and Leios.
+
+ALBA is a generic tool that lets one prove knowledge of any kind of data
+elements, but
+as noted in (todo: alba applications section), these three applications
+don't need the full expressivity of ALBA, they really need
+a multisignature scheme, where the data elements would specifically be
+signatures.
+Thus there is question of whether we want our library to provide a general
+ALBA interface or simply an interface for a multisignature scheme.
+We will do both.
+1. We will implement the most generic version of ALBA to make it suitable for
+as many applications as possible.
+This means that there is only a generic notion of "data elements" and no notion
+of signatures, VRF, etc.
+2. We will use the generic ALBA library to implement a multisignature scheme,
+either as a small example in this repository or a full-fledged library in a
+separate repository.
+Either way, the multi-signature implementation would serve as a flagship
+example of how the ALBA library can be used.
+
+The requirements for our (general) ALBA implementation can be roughly split
+in four groups: 
+1. definitions: what do the ALBA schemes we implement guarantee, under what
+assumptions?
+2. correctness: does our implementation provide the same guarantees of
+soundness and completeness as the ALBA schemes on paper?
+3. performance: proof size, communication complexity, computational complexity.
+4. interfaces: the applications should be able to use our
+implementation relatively easily
+
+### Definitions
+
+The definitions requirement can be stated as follows.
+Taking an ALBA protocol we implement, can you construct (on paper) a larger
+protocol, such as Mithril, and mathematically prove its guarantees such as
+completeness and soundness?
+Currently, ALBA schemes satisfying the definitions in the ALBA paper cannot
+be used to construct Mithril, Peras and Leios.
+
+First, all three require a multisignature scheme secure against
```

Comment:
Remove the following, whole paragraphs. We want to cover these cases, not focus on it. You can say much more succinctly what you want without using them as points.